### ISSUE TITLE
Action noise in testing state

### ISSUE THREAD

**Issue by luyiming** [thu-ml/tianshou#304]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [x] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:

I find that action noise is added both in training and testing phase. For example, the following code is from `DDPGPolicy.forward`. Is there any reason to add action noise when you are evaluating the policy? And what is the best practice if I wish a deterministic policy evaluation?

  ```python
    def forward(
        self,
        batch: Batch,
        state: Optional[Union[dict, Batch, np.ndarray]] = None,
        model: str = "actor",
        input: str = "obs",
        **kwargs: Any,
    ) -> Batch:
        model = getattr(self, model)
        obs = batch[input]
        actions, h = model(obs, state=state, info=batch.info)
        actions += self._action_bias
        if self._noise and not self.updating:
            actions += to_torch_as(self._noise(actions.shape), actions)
        actions = actions.clamp(self._range[0], self._range[1])
        return Batch(act=actions, state=h)
  ```

**Comment by ChenDRAG** [comment]
@luyiming This indeed is a bug for DDPG policy in older versions, but it has been fixed by #280 , and merged into master in the latest version of Tianshou.(tag 0.4.0) However, 0.4.0 has refactored tianshou codebase and changed quite a lot of APIs. You will need some time to get familar with the new version, very sorry about that.

An alternative way to avoid switching to newest version and also solve "evaluating noise" would be to avoid pass in Guass noise into DDPG policy and only use expl_noise option in train collector.

**Comment by luyiming** [comment]
Thanks for your reply! I have checked the latest version and I think the new code structure is easier to understand.
